ArcMovies/ArcMovies/Services/TMDbApiService.cs

```diff
@@ -0,0 +1,149 @@
+                    {
+                        using (var responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                        {
+                            return JsonConvert.DeserializeObject<SearchMovie>(
+                                await new StreamReader(responseStream).ReadToEndAsync().ConfigureAwait(false));
+                        }
+                    }
+                }                    
+            }
+            catch (Exception ex)
+            {
+                ReportError(ex);
+            }            
+
+            return null;
+        }
+                
+        public async Task<SearchMovie> GetMoviesByCategoryAsync(int page, Enums.MovieCategory category)
+        {
+            var restUrl = $"{ApiUrl}{Enums.PathCategoryMovie(category)}?api_key={ApiKey}&page={page}&language={_language}";
+            try
+            {
+                using (var response = await _httpClient.GetAsync(restUrl).ConfigureAwait(false))
+                {
+                    if (response.IsSuccessStatusCode)
+                    {
+                        using (var responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                        {
+                            return JsonConvert.DeserializeObject<SearchMovie>(
+                                await new StreamReader(responseStream).ReadToEndAsync().ConfigureAwait(false));
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                ReportError(ex);
+            }            
+
+            return null;
+        }
+        
+        public async Task<MovieDetail> GetMovieDetailAsync(int id)
+        {
+            var restUrl = $"{ApiUrl}{MoviesPath}/{id}?api_key={ApiKey}&language={_language}";
+            try
+            {
+                using (var response = await _httpClient.GetAsync(restUrl).ConfigureAwait(false))
+                {
+                    if (response.IsSuccessStatusCode)
+                    {
+                        using (var responseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                        {
+                            return JsonConvert.DeserializeObject<MovieDetail>(
+                                await new StreamReader(responseStream).ReadToEndAsync().ConfigureAwait(false));
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                ReportError(ex);
+            }            
+
+            return null;
+        }
+        
+        public async Task<List<Genre>> GetGenresAsync()
+        {            
+            var restUrl = $"{ApiUrl}{GenreListPath}?api_key={ApiKey}&language={_language}";
+            try
+            {
+                using (var response = await _httpClient.GetAsync(restUrl).ConfigureAwait(false))
```

Comment:
All methods that make a rest call have basically the same body, only differing the url and the type used to convert the response content to an object. It would be good to have a generic method that would take the url as 
parameter for reuse.